private/buf/buffetch/format.go

```diff
@@ -14,6 +14,33 @@
 
 package buffetch
 
+// TODO: remove the unexported formats below and only use these ones.
+// A good time to do so is when we merge txtpb and binpb into this branch.
+const (
+	// formatBin is the binary format.
+	FormatBin = "bin"
```

Comment:
Optimally, these should not be public - the idea is all of this is hidden within `buffetch`. Exposing these breaks the `buffetch` model. I don't have any great answers, but it feels like we're really exposing `buffetch` internals too much. Perhaps this is the best we can easily do.

Reply:
This is the easiest way I found. The two alternatives I considered are:
1. Return a `RawRef`, which break the API even more, IMO
2. Lifted types like `ParsedGitRef` to `buffetch` from `buffetch.internal` and expose them. This also breaks the existing API of `bufftetch`.